src/transformers/models/mobilevit/modeling_mobilevit.py

```diff
@@ -0,0 +1,1083 @@
+                    in_channels=in_channels,
+                    out_channels=out_channels,
+                    kernel_size=3,
+                    dilation=rate,
+                    use_act="relu",
+                )
+                for rate in config.atrous_rates
+            ]
+        )
+
+        pool_layer = MobileViTASPPPooling(config, in_channels, out_channels)
+        self.convs.append(pool_layer)
+
+        self.project = MobileViTConvLayer(
+            config, in_channels=5 * out_channels, out_channels=out_channels, kernel_size=1, use_act="relu"
+        )
+
+        self.dropout = nn.Dropout(p=config.aspp_dropout_prob)
+
+    def forward(self, features: torch.Tensor) -> torch.Tensor:
+        pyramid = []
+        for conv in self.convs:
+            pyramid.append(conv(features))
+        pyramid = torch.cat(pyramid, dim=1)
+
+        pooled_features = self.project(pyramid)
+        pooled_features = self.dropout(pooled_features)
+        return pooled_features
+
+
+class MobileViTDeeplabV3(nn.Module):
+    """
+    DeepLabv3 architecture: https://arxiv.org/abs/1706.05587
+    """
+
+    def __init__(self, config: MobileViTConfig) -> None:
+        super().__init__()
+        self.aspp = MobileViTASPP(config)
+
+        self.classifier = nn.Sequential()
+        self.classifier.add_module(name="classifier_dropout", module=nn.Dropout2d(config.classifier_dropout_prob))
+        self.classifier.add_module(
+            name="classifier",
+            module=MobileViTConvLayer(
+                config,
+                in_channels=config.aspp_out_channels,
+                out_channels=config.num_labels,
+                kernel_size=1,
+                use_norm=False,
+                use_act=False,
+                bias=True,
+            ),
+        )
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        features = self.aspp(hidden_states[-1])
+        features = self.classifier(features)
+        return features
+
+
+@add_start_docstrings(
+    """
+    MobileViT model with a semantic segmentation head on top, e.g. for Pascal VOC.
+    """,
+    MOBILEVIT_START_DOCSTRING,
+)
+class MobileViTForSemanticSegmentation(MobileViTPreTrainedModel):
+    def __init__(self, config: MobileViTConfig) -> None:
+        super().__init__(config)
+
+        self.num_labels = config.num_labels
+        self.mobilevit = MobileViTModel(config, expand_output=False)
+        self.seg_head = MobileViTDeeplabV3(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def compute_loss(self, logits, labels):
+        # upsample logits to the images' original size
+        upsampled_logits = nn.functional.interpolate(
+            logits, size=labels.shape[-2:], mode="bilinear", align_corners=False
+        )
+        loss_fct = CrossEntropyLoss(ignore_index=self.config.semantic_loss_ignore_index)
+        loss = loss_fct(upsampled_logits, labels)
+        return loss
```

Comment:
@sgugger isn't a fan of computing the loss in a separate function, so you can directly include it in the forward method (think we also do this for other semantic segmentation models).

Reply:
Will change. Note that I copied this from BEiT (minus the aux loss).